Automattic/wp-calypso
# Add LinkedIn to social-previews package

Related to 1203820933396971-as-1204281808721018/f

## Proposed Changes

* Create `LinkedInPreview` component in `social-previews` package
* Use `LinkedInPreview` in WPCOM posts share preview at `https://wordpress.com/posts/:site`



### Prerequisites
- Make sure you have a LinkedIn account.
- Make sure you have a self-hosted site connected to Jetpack, with a subscription that includes Social.
- From the _Social_ or _My Jetpack_ page of your site WPadmin, activate Social.
- Then, connect your site to LinkedIn (check [these instructions](https://jetpack.com/support/jetpack-social/connecting-to-social-networks/) if you need help). You may need to create a test page.
- Spin up Calypso, by running this branch locally or using a live link below.

### Testing

- Create a new post from your site `wp-admin`
- Add a featured image (do not add a custom media)
- Publish it
- Visit `/posts/:site` in Calypso
- You should see the newly created post under the _Published_ tab
- In the post menu, click _Share_, then press the _Preview_ button
- Select the LinkedIn preview
- You should see a preview of how the post looks when shared
<img width="1419" alt="Screenshot 2023-04-18 at 2 46 25 PM" src="https://user-images.githubusercontent.com/18226415/232736521-3f526cc9-0693-4c83-9aa2-5a514339f745.png">


When used in Jetpack Social preview (PR coming soon)
<img width="1283" alt="Screenshot 2023-04-18 at 1 11 00 PM" src="https://user-images.githubusercontent.com/18226415/232736670-e375dc68-3e18-403c-8963-0e98b556ddd3.png">
<img width="1294" alt="Screenshot 2023-04-18 at 1 12 32 PM" src="https://user-images.githubusercontent.com/18226415/232736683-253afddc-af1b-435b-bfb8-50e574cf5d2a.png">


## Pre-merge Checklist




---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - 0-as-1204281808721018
